.github/workflows/lint_and_integration.yaml

```diff
@@ -0,0 +1,140 @@
+name: Lint and integration tests
+
+on:
+  push:
+    paths-ignore:
+      - 'docs/**'
+    branches:
+      - main
+      - autoupdate/strict
+      - autoupdate/moonray
+      - 'release-[0-9]+.[0-9]+'
+      - 'autoupdate/release-[0-9]+.[0-9]+-strict'
+      - 'autoupdate/sync/**'
```

Comment:
I wonder why we are so explicit about the autoupdate branches, can you remember @berkayoz?
Why don't we just use
```
  - main
  - autoupdate/**                
  - 'release-[0-9]+\.[0-9]+'  
  ```

Reply:
I'm not sure about the why but personally being more explicit/expecting a pattern feels less prone to errors. Up to you folks

Reply:
Discussed offline `autoupdate/sync` is obsolete.
@petrutlucian94 feel free to either use `autoupdate/**` or `autoupdate/moonray`, ... but remove the `autoupdate/sync` in any case